modules/core/src/main/java/org/apache/ignite/internal/IgnitionEx.java

```diff
@@ -1472,6 +1477,32 @@ private static void notifyStateChange(@Nullable String igniteInstanceName, Ignit
             lsnr.onStateChange(igniteInstanceName, state);
     }
 
+    /**
+     * @return map of resources for overriding and clear the map after it.
+     */
+    public static Map<String, WrappableResource> takeTestResources() {
+        Map<String, WrappableResource> resources = Optional.ofNullable(testResources.get()).orElse(Collections.EMPTY_MAP);
+
+        testResources.remove();
+
+        return resources;
+    }
+
+    /**
+     * @param rsrc Resource.
+     */
+    public static void addTestResource(WrappableResource rsrc) {
```

Comment:
This method has tricky semantics: resources will be injected to next node started by current thread. We should explain it in javadoc.